### DISCUSSION TITLE
Same as SFTP extension for VS Code ?

### DISCUSSION THREAD

**Discussion by avanhovedev** [zed-industries/zed#12082] [Feature Requests]
Do you think it would be a good idea to think about an SFTP extension like the one from VSCode? https://marketplace.visualstudio.com/items?itemName=Natizyskunk.sftp

In my case, yes, because I like to work locally and then push the code to my remote server.

**Comment by SomeoneToIgnore** [comment]
> and then push the code to my remote server.

If all you care about is some SFTP uploading command have a CLI tool working already, you might consider using tasks: https://zed.dev/docs/tasks

Zed can store a definition of that "upload to SFTP" bash invocation generic over project path and a few other things, and launch that via a single shortcut/task modal invocation.

As for the full extension, currently that would be rather hard, as this is what any Zed extension is limited with:
https://github.com/zed-industries/zed/blob/d5fe2c85d80288a0f50eb1c2dfb1e82f11046402/crates/extension_api/src/extension_api.rs#L59-L118

and the extensions run in wasm runtime, isolated from most of the world by default.

**Reply by avanhovedev** [reply]
Thanks for the answer !

**Comment by tjohnman** [comment]
Did you find a solution you were satisfied with? This has become an integral part of my workflow, and in trying Zed it has become a stumbling block for me.

**Reply by cukabeka** [reply]
I would be interested in a SFTP Sync like the mentioned VS extension, too. Plus, from how I understood @SomeoneToIgnore, this would only upload, not sync like the VS extension. So still not sure, if this works well.

**Comment by andreyc0d3r** [comment]
Help me build it, I need the sftp functionality too. I got it to work so far, but I have limited time to test all the features.

https://github.com/andreyc0d3r/zed-sftp

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
